resources/party.ts

```diff
@@ -14,6 +15,27 @@ const emptyRoleToPartyNames = () => {
   };
 };
 
+export type BasePartyMemberParamObject = {
+  role?: Role;
+  job?: Job;
+  id?: string;
+  name: string;
+  nick: string;
+};
+
+export type PartyMemberParamObjectKeys = keyof BasePartyMemberParamObject;
+
+export interface PartyMemberParamObject
+  extends OutputStringsParamObject, BasePartyMemberParamObject {
+  toString: () => string;
+}
+
+// This is a partial interface of both RaidbossOptions and OopsyOptions.
+export interface PartyTrackerOptions {
+  DefaultPlayerLabel: keyof PartyMemberParamObject;
+  PlayerNicks: { [gameName: string]: string };
+}
```

Comment:
If this is meant to be a partial interface then it may make sense to use `Pick` to enforce that contract? Something like this?

```ts
export type PartyTrackerOptions = Pick<RaidbossOptions, 'DefaultPlayerLabel'> & {
  DefaultPlayerLabel: keyof PartyMemberParamObject;
}
```

Reply:
It is meant to be a partial interface, but it's more that those two interfaces extend (implicitly) this one.

I feel like the contract is already enforced by typescript as if either of those options change to be not exist or be incompatible, then this callsite will break.

Reply:
Ok, I added an explicit type here. Because this is TypeScript and it's all duck typing this felt ok to me originally, but you're right that being explicit is better here.